pkg/store/postings_codec.go

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+package store
+
+import (
+	"bytes"
+
+	"github.com/golang/snappy"
+	"github.com/pkg/errors"
+	"github.com/prometheus/prometheus/tsdb/encoding"
+	"github.com/prometheus/prometheus/tsdb/index"
+)
+
+const (
+	// These headers should not be a prefix of each other.
+	codecHeaderRaw    = "diff+varint+raw"
```

Comment:
```suggestion
	codecHeaderRaw    = "diff+varint"
```
?

Reply:
This is exactly what would not work ;-) Since we compare prefix of the input byte array, HasPrefix(input, "diff+varint") returns true for both snappy and not snappy version. I wanted prefix to be unique.

Reply:
Ok, now is the question why do we need both. Can we only support snappy?

Reply:
Yes, you're right.